File: src/main/java/com/jss/osiris/modules/osiris/reporting/service/IncidentResponsibilityService.java
package com.jss.osiris.modules.osiris.reporting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.reporting.model.IncidentResponsibility;

public interface IncidentResponsibilityService {
    public List<IncidentResponsibility> getIncidentResponsibilities();

    public IncidentResponsibility getIncidentResponsibility(Integer id);

    public IncidentResponsibility addOrUpdateIncidentResponsibility(IncidentResponsibility incidentResponsibility);
}

File: client/src/app/modules/reporting/services/incident.responsibility.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { IncidentResponsibility } from '../../reporting/model/IncidentResponsibility';

@Injectable({
  providedIn: 'root'
})
export class IncidentResponsibilityService extends AppRestService<IncidentResponsibility>{

  constructor(http: HttpClient) {
    super(http, "reporting");
  }

  getIncidentResponsibilities() {
    return this.getList(new HttpParams(), "incident-responsibilities");
  }
  
   addOrUpdateIncidentResponsibility(incidentResponsibility: IncidentResponsibility) {
    return this.addOrUpdate(new HttpParams(), "incident-responsibility", incidentResponsibility, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
